pkg/rpc/core/mempool.go

```diff
@@ -27,43 +25,23 @@ const subscribeTimeout = 5 * time.Second // TODO: Make configurable?
 // More: https://docs.cometbft.com/v0.37/rpc/#/Tx/broadcast_tx_async
 func BroadcastTxAsync(ctx *rpctypes.Context, tx cmttypes.Tx) (*ctypes.ResultBroadcastTx, error) {
 	unwrappedCtx := ctx.Context()
-
-	var res *abci.ResponseCheckTx
-	responseCh := make(chan *abci.ResponseCheckTx, 1)
-
-	err := env.Adapter.Mempool.CheckTx(tx, func(response *abci.ResponseCheckTx) {
-		responseCh <- response
-	}, mempool.TxInfo{})
+	err := env.Adapter.Mempool.CheckTx(tx, nil, mempool.TxInfo{})
 	if err != nil {
-		return nil, fmt.Errorf("error during CheckTx: %w", err)
+		return nil, err
 	}
-
-	// Wait for the callback to be called
-	select {
-	case res = <-responseCh:
-		// Successfully received CheckTx response
-	case <-unwrappedCtx.Done():
-		return nil, fmt.Errorf("context cancelled while waiting for CheckTx response: %w", unwrappedCtx.Err())
+	// gossipTx optimistically
+	err = env.Adapter.TxGossiper.Publish(unwrappedCtx, tx)
+	if err != nil {
+		return nil, fmt.Errorf("tx added to local mempool but failed to gossip: %w", err)
```

Comment:
Remove TX from mempool or it can not be submitted again

Reply:
Agree here. If it passed checktx it could stick in the mempool and never be gossiped to the proposer. Should we retry or remove it from the app mempool somehow?